samples-generator/lib/createPointCloudTile.js

```diff
@@ -671,76 +824,156 @@ function getColorsRGBA(colors) {
         buffer.writeUInt8(g, i * 4 + 1);
         buffer.writeUInt8(b, i * 4 + 2);
         buffer.writeUInt8(a, i * 4 + 3);
+
+        minComp[0] = Math.min(minComp[0], r);
+        minComp[1] = Math.min(minComp[1], g);
+        minComp[2] = Math.min(minComp[2], b);
+        minComp[3] = Math.min(minComp[3], a);
+        maxComp[0] = Math.max(maxComp[0], r);
+        maxComp[1] = Math.max(maxComp[1], g);
+        maxComp[2] = Math.max(maxComp[2], b);
+        maxComp[3] = Math.max(maxComp[3], a);
     }
     return {
         buffer : buffer,
-        propertyName : 'RGBA',
+        propertyName : use3dTilesNext ? 'COLOR_0' : 'RGBA',
         componentType : 'UNSIGNED_BYTE',
-        type : 'VEC4'
+        type : 'VEC4',
+        min : minComp,
+        max : maxComp,
+        count : colorsLength,
+        normalized: true,
+        byteStride: 4
     };
 }
 
 function getColorsRGB565(colors) {
     var colorsLength = colors.length;
     var buffer = Buffer.alloc(colorsLength * sizeOfUint16);
+
+    var minComp = [Number.POSITIVE_INFINITY];
+    var maxComp = [Number.NEGATIVE_INFINITY];
+
     for (var i = 0; i < colorsLength; ++i) {
         var color = colors[i];
         var r = Math.floor(color.red * 31); // 5 bits
         var g = Math.floor(color.green * 63); // 6 bits
         var b = Math.floor(color.blue * 31); // 5 bits
         var packedColor = (r << 11) + (g << 5) + b;
+        minComp[0] = Math.min(minComp[0], packedColor);
+        maxComp[0] = Math.max(maxComp[0], packedColor);
         buffer.writeUInt16LE(packedColor, i * sizeOfUint16);
     }
     return {
         buffer : buffer,
-        propertyName : 'RGB565',
+        propertyName :  'RGB565',
         componentType : 'UNSIGNED_SHORT',
-        type : 'SCALAR'
+        type : 'SCALAR',
+        min : minComp,
+        max : maxComp,
+        count : colorsLength
     };
 }
 
-function getBatchTableForBatchedPoints(batchLength) {
+function getBatchTableForBatchedPoints(batchLength, use3dTilesNext) {
     // Create some sample per-batch properties. Each batch will have a name, dimension, and id.
     var names = new Array(batchLength); // JSON array
     var dimensionsBuffer = Buffer.alloc(batchLength * 3 * sizeOfFloat32); // Binary
-    var idBuffer = Buffer.alloc(batchLength * sizeOfUint32); // Binary
+    var idBuffer;
+    if (!use3dTilesNext) {
+        idBuffer = Buffer.alloc(batchLength * sizeOfUint32); // Binary
+    }
```

Comment:
For both batched points and per-property points I think it's fine to keep the id's. The naming can be misleading, but these are not the same as the feature ids stored in the vertex attribute. I noticed this when diffing files between this branch and master.

Reply:
Keeping the IDbuffer